### ISSUE TITLE
Polylang support for ACF Options

### ISSUE THREAD

**Issue by pvtryk** [wp-graphql/wpgraphql-acf#63]
### Description

I am using Polylang plugin to manage a multilingual website.

When I write a query to get the data from the options page with language param I get the wrong data.
I get the correct data when I disable the "ACF Options for Polylang" plugin and remove the language param from the query.

### Steps to reproduce

1. Go to GraphQL IDE inside Wordpress.
2. Write a query with language param
```
query example {
  themeSettings(language: EN) {
    optionsPage {
      login {
        headline
        description
      }
    }
  }
}
```
3. You won't get the correct data.

### WPGraphQL Version

1.14.4

### WPGraphQL For ACF Version

2.0.0-beta.3.1.0

### ACF (Advanced Custom Fields) Version. Free or Pro?

Pro

### WordPress Version

6.2.2

### PHP Version

8.2

### Additional enviornment details

Plugins: 
1. Polylang Pro (3.3.2) Author: WP SYNTEX
2. Polylang for Woocommerce (1.7.2) Author: WP SYNTEX
3. ACF Options for Polylang (1.1.10) Author: BE Api
4. ACF Pro (6.1.6)

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL, WPGraphQL For ACF, ACF, etc.

- [X] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by jasonbahl** [comment]
@pvtryk do you know where the `language` argument here: `themeSettings(language: EN) {` is coming from?

The argument is not added by this plugin, so another codebase is adding that and would be responsible for overriding how resolvers should execute when that argument is present.

**Comment by cristobalbahe** [comment]
Hello everyone,

I think it comes from the plugin wp-graphql-polylang (https://github.com/valu-digital/wp-graphql-polylang), as I am running into the same problem.

All translations do work with the new wpgraphql-acf plugin except for the options page that won't return data of the given language.

**Comment by robinjessome** [comment]
> do you know where the `language` argument here: `themeSettings(language: EN)` is coming from?

Pretty sure it's the [ACF Options for Polylang](https://github.com/BeAPI/acf-options-for-polylang) that's enabling the language argument.

**Comment by jasonbahl** [comment]
If someone can provide very clear steps to reproduce that would go a  long way.

ideally:
- an export of ACF Field Group(s) in use
- version(s) of all plugins (and versions) installed and activated
- queries you're executing
- expectation vs actual results

The more info provided, the better we can help get to a resolution

Also, if you've done any digging so far as to how Polylang was working with the old WPGraphQL for ACF, that would be cool to share too. I'm not sure how Polylang was supporting this in the past, but if you have code you can link to shat shows how it worked with the old WPGraphQL for ACF, then porting support to the new one will be faster

In short, the more info, the better and the faster we can reproduce and resolve. 🙏🏻

**Comment by robinjessome** [comment]
### Plugins: 
- Advanced Custom Fields PRO (v6.2.6)
- Polylang (v3.5.4)
- WPGraphQL (v1.20.0)
- WP GraphQL Polylang (v0.6.0)
- ACF Options for Polylang (v1.1.11)

ACF / WPGraphQL plugins: 
- WPGraphQL for Advanced Custom Fields (v0.6.1) WORKS
- WPGraphQL for ACF (v2.1.0) Does not work

### ACF Fields/Options page export:
[acf-export-2024-02-08.json](https://github.com/wp-graphql/wpgraphql-acf/files/14210413/acf-export-2024-02-08.json)
Includes options page and a basic field for it. (Requires ACF Pro)

### Theme customization
To enable ACF Options pages in GraphQL with the classic v0.6.1 plugin, add this in your theme `functions.php` file.
This is done in the dashboard with the new v2.1.0 plugin.

```php
add_filter('acf/get_options_page', function ($page, $slug) {;
  
  if($slug === 'testoptions') {
      $page['show_in_graphql'] = true;
      $page['graphql_field_name'] = 'testOptions';
  }
  return $page;
}, 10, 2);
```

### Query the options page
```graphql
query GetTestOptionsPageClassic {
  english: testOptions(language: EN) {
    optionsFields {
      siteTitle
    }
  }
  french: testOptions(language: FR) {
    optionsFields {
      siteTitle
    }
  }
}
```
With the **classic** plugin enabled, the `language` argument the translated content is fetched as expected.
![Screenshot 2024-02-08 at 9 40 57 AM](https://github.com/wp-graphql/wpgraphql-acf/assets/6805734/5ae10535-1ccb-4534-bee7-21e219d5a4ff)

WIth the **new** plugin enables and the ACF Options for Polylang plugin disabled, it returns only the main field (assuming it's been filled) since there are no translations)

![Screenshot 2024-02-08 at 9 40 31 AM](https://github.com/wp-graphql/wpgraphql-acf/assets/6805734/b7ff2771-a363-41fa-89b6-c1c0e6509118)

**Comment by jasonbahl** [comment]
@robinjessome I'm unable to get the `language` field to show up on options pages when using the _old_ WPGraphQL for ACF 🤔 

Here's what I've done to try and reproduce: 

### Active Plugins: 

- ACF Pro v6.2.4
- ACF Options for Polylang v1.1.11
- Polylang v3.5.4
- WP GraphQL Polylang v0..6.0
- WPGraphQL v1.21.0
- WPGraphQL for ACF v0.6.1 (and also tried 2.1.2)

### Register ACF Options Page using ACF Pro's UI

![Image](https://github.com/wp-graphql/wpgraphql-acf/assets/1260765/986a9219-16f5-401d-8cc0-c0fc6247b9c7)

For WPGraphQL v0.6.1, I added the following Snippet to show the Options page in WPGraphQL: 

```php
add_filter('acf/get_options_page', function ($page, $slug) {;
  
  if($slug === 'polylang-test-options-page') {
      $page['show_in_graphql'] = true;
      $page['graphql_field_name'] = 'PolylangTestOptionsPage';
  }
  return $page;
}, 10, 2);
```

### Register an ACF Field Group to show on the options page 


![Image](https://github.com/wp-graphql/wpgraphql-acf/assets/1260765/c915cf59-15be-4632-bf45-da1fa1c04661)

Attached is the .json export of the ACF Field Group and ACF Options Page

[acf-export-2024-02-22 (2).json.zip](https://github.com/wp-graphql/wpgraphql-acf/files/14366612/acf-export-2024-02-22.2.json.zip)


Now, with WPGraphQL for ACF v0.6.1 or v2.1.2 active, I'm able to see `polylangTestOptionsPage` as a root field in the schema, but I do not see a "language" argument on the field.

I do see the "language" argument on many fields in the Schema:

![Image](https://github.com/wp-graphql/wpgraphql-acf/assets/1260765/683a6028-a059-43e9-8bc9-6eb5515ab76c)


But not the options page nor the options page field group:

![Image](https://github.com/wp-graphql/wpgraphql-acf/assets/1260765/e1aa5ffc-6275-4cdc-97c9-ec1ab225bcd8)

----

This leads me to believe that I'm missing a step to reproduce. Perhaps a setting somewhere? Or another code snippet?

Are you able to reproduce this on a vanilla WordPress install following the same steps I have followed? Am I missing any steps? Is there any custom code in a theme or custom plugin that's filtering the `language` argument onto the options page(s) and filtering the resolver(s)?

**Comment by jasonbahl** [comment]
@robinjessome ah ha! 

I think I found the code that bridged polylang -> ACF: https://github.com/valu-digital/wp-graphql-polylang/blob/master/src/OptionsPages.php#L8-L9

fwiw, it doesn't work for me with WPGraphQL for ACF v0.6.1 🤔 but at least I might have a starting point to look into 👀

**Comment by jasonbahl** [comment]
Currently supporting a cross-over between ACF and polylang is out of scope for WPGraphQL for ACF. 

What I would recommend is either contributing a Pull Request to WPGraphQL Polylang to [update their code](https://github.com/valu-digital/wp-graphql-polylang/blob/master/src/OptionsPages.php#L8-L9) to be compatible with the latest (and/or multiple versions) of WPGraphQL for ACF and/or implement similar code in your own projects, or come up with and publish a new specific "bridge" plugin intended on bridging WPGraphQL for ACF + Polylang. 

If/when a solution is developed, I would recommend opening a Pull Request to this repo with some tests to ensure the filters/hooks/functions called in said solution continue to work in the future. 

I believe ultimately a regression was likely caused by something in core WPGraphQL or WPGraphQL for ACF that lead to the code in WPGraphQL Polylang to no longer work, and we clearly didn't have a test case in place to prevent that regression, so if / when a solution is developed, please contribute a test case, or at minimum provide a link to updated functioning code so we can ensure the use case is tested against. 🙏 

I'm going to close this as officially out-of-scope, but I hope the research I did and the links I provided above can help get ya'll into a good spot with the bridging of multiple plugins.
